packages/algorand-js/src/internal/interpreter/opcode-list.ts

```diff
@@ -0,0 +1,177 @@
+  }
+}
+
+// pops two unit64 from stack(a, b) and pushes their diff(a - b) to stack
+// panics on underflow (result < 0)
+export class Sub extends Op {
+  execute (stack: IStack<string | bigint>): void {
+    this.assertStackLen(stack, 2);
+    const a = stack.pop();
+    const b = stack.pop();
+    if (typeof a !== 'bigint' || typeof b !== 'bigint') {
+      throw new TealError(ERRORS.TEAL.INVALID_OPERATION, {
+        opcode: "-"
+      });
+    }
+    this.checkUnderFlow(a - b);
+    stack.push(a - b);
+  }
+}
+
+// pops two unit64 from stack(a, b) and pushes their division(a / b) to stack
+// panics if b == 0
+export class Div extends Op {
+  execute (stack: IStack<string | bigint>): void {
+    this.assertStackLen(stack, 2);
+    const a = stack.pop();
+    const b = stack.pop();
+    if (typeof a !== 'bigint' || typeof b !== 'bigint' || b === BigInt(0)) {
+      throw new TealError(ERRORS.TEAL.INVALID_OPERATION, {
+        opcode: "/"
+      });
+    }
+    stack.push(a / b);
+  }
+}
+
+// pops two unit64 from stack(a, b) and pushes their mult(a * b) to stack
+// panics on overflow (result > max_unit64)
+export class Mul extends Op {
+  execute (stack: IStack<string | bigint>): void {
+    this.assertStackLen(stack, 2);
+    const a = stack.pop();
+    const b = stack.pop();
+    if (typeof a !== 'bigint' || typeof b !== 'bigint') {
+      throw new TealError(ERRORS.TEAL.INVALID_OPERATION, {
+        opcode: "*"
+      });
+    }
+    this.checkOverFlow(a * b);
+    stack.push(a * b);
+  }
+}
+
+// pushes 1st argument from argument array to stack
+export class Arg_0 extends Op {
+  private readonly _args;
+
+  constructor (args: Array<string | number>) {
+    super();
+    this._args = args;
+  };
+
+  execute (stack: IStack<string | bigint>): void {
+    const arg = this._args[0];
+    if (arg === undefined || (typeof arg !== 'string' && typeof arg !== 'number')) {
+      throw new TealError(ERRORS.TEAL.INVALID_OPERATION, {
+        opcode: "arg_0"
+      });
+    }
+    if (typeof arg === 'number') {
+      this.checkOverFlow(BigInt(arg));
+      stack.push(BigInt(arg));
+    } else { stack.push(arg); }
+  }
+}
+
+// pushes 2nd argument from argument array to stack
+export class Arg_1 extends Op {
+  private readonly _args;
+
+  constructor (args: Array<string | number>) {
+    super();
+    this._args = args;
+  };
+
+  execute (stack: IStack<string | bigint>): void {
+    const arg = this._args[1];
+    if (arg === undefined || (typeof arg !== 'string' && typeof arg !== 'number')) {
+      throw new TealError(ERRORS.TEAL.INVALID_OPERATION, {
+        opcode: "arg_1"
+      });
+    }
+    if (typeof arg === 'number') {
+      this.checkOverFlow(BigInt(arg));
+      stack.push(BigInt(arg));
+    } else { stack.push(arg); }
+  }
+}
+
+// pushes 3rd argument from argument array to stack
+export class Arg_2 extends Op {
+  private readonly _args;
+
+  constructor (args: Array<string | number>) {
+    super();
+    this._args = args;
+  };
```

Comment:
this class can inherit from `Arg_1`. Then we don't need to declare the constructor nor the `_args`,

Reply:
Alright